tests/publiccloud/slem_upgrade_next.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 - 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: in public cloud, upgrade SLE Micro to 6.2+
+#
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use publiccloud::utils qw(is_byos is_azure is_ec2 trup_call_remote);
+use utils qw(zypper_call systemctl);
+use version_utils qw(is_sle_micro is_micro);
+use transactional 'check_target_version';
+
+sub run {
+    my ($self, $args) = @_;
+    return unless (is_sle_micro('6.1+'));
+
+    my $instance = $self->{my_instance} = $args->{my_instance};
+    my $timeout = 900;
+
+    # start migration
+    select_serial_terminal();
+    $instance->ssh_script_retry("sudo zypper -n ref", retry => 3, timeout => int($timeout / 3), fail_message => "zypper refresh failed");
+    record_info('Repos', $instance->ssh_script_output('zypper lr -u'));
+    $instance->ssh_update_transactional_system();
+    my ($out, $exit) = $instance->trup_call_remote("migration", timeout => $timeout);
```

Comment:
If we're not using the return values, we can further simplify the function to avoid returning 2 values.

Reply:
> This is an opinion

This is not an opinion.  It's called best practices when designing libraries.